lambdas/postgres-migration-count-tool/README.md

```diff
@@ -0,0 +1,88 @@
+# Postgres Migration Count Tool Lambda
+
+## This lambda runs a script that:
+
+1) Scans DynamoDB AsyncOperations, Rules, Executions and Collections and compares their record count with the configured postgres database.
+2) Builds a mapping of collections in the DynamoDb `collections` databases to `collections` in the Postgres database, notes any that are missing and then compares `granules`, `executions` and `pdrs` between the Elasticsearch database replication and the Postgres database.
+3) Returns a report in the format:
+
+```text
+collectionsNotMapped: {collectionFailures[]},
+records_in_dynamo_not_in_postgres: {aggregateReportObj},
+pdr_granule_and_execution_records_not_in_postgres_by_collection: {collectionReportObj},
+s3Uri: 's3://uri'
+```
+
+Where
+
+* `collectionFailures` -- an Error object with a serialized .collection attached
+* `records_in_dynamo_not_in_postgres` -- an object containing total Dynamo record counts, and the delta vs Postgres:
+ {
+    collectionsDelta: number;
+    totalDynamoCollections: number;
+    providersDelta: number;
+    totalDynamoProviders: number;
+    rulesDelta: number;
+    totalDynamoRules: number;
+    asyncOperationsDelta: number;
+    totalDynamoAsyncOperations: number;
+}
+* `pdr_granule_and_execution_records_not_in_postgres_by_collection` -- an object containing a set of collection keys containing count deltas and the dynamo totals for each collection:
+{
+  <collection> {
+        pdrsDelta: number;
+        totalPdrs: number;
+        granulesDelta: number;
+        totalGranules: number;
+        executionsDelta: number;
+        totalExecutions: number;
+    };
+}
+* s3Uri -- If the `reportBucket` and `reportKey` params are provided to the Lambda/API endpoint, the storage location on S3 of the output report.
+
+## Running the Report
+
+This tool can be invoked one of two ways:
+
+### Via direct lambda invocation:
+
+```bash
+aws lambda invoke --function-name $PREFIX-postgres-migration-count-tool --payload $PAYLOAD $OUTFILE
+```
+This will invoke the lambda synchronously.  Please note that depending on your data holdings, this may exceed the 15 minute AWS Lambda limit, if this occurs, you will need to invoke the tool via the API as an asynchronous operation.
+
+Where:
+
+* PAYLOAD - base64 encoded JSON object.   For example:
+
+```bash
+--payload $(echo '{"reportBucket": "cumulus-test-sandbox-internal", "reportPath": "jk-tf4/testOutput"}' | base64)
```

Comment:
Is this actually a valid example given it only includes 2 of the 5 parameters listed below?

Reply:
Hmm, probably.    Those two parameters are the most likely to be used by end users, adding all of them felt like it contributed more noise than needed, given the list could change, etc.

Reply:
That said, b1e9e6831